src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -398,6 +402,43 @@ public ResponseEntity<byte[]> getAttachmentUnitAttachment(@PathVariable Long att
         return buildFileResponse(Path.of(FilePathService.getAttachmentUnitFilePath(), String.valueOf(attachmentUnit.getId())).toString(), filename);
     }
 
+    /**
+     * GET files/attachments/slides/attachment-unit/:attachmentUnitId/slide/:slideNumber : Get the lecture unit attachment slide by slide number
+     *
+     * @param attachmentUnitId ID of the attachment unit, the attachment belongs to
+     * @param slideNumber      the slideNumber of the file
+     * @return The requested file, 403 if the logged-in user is not allowed to access it, or 404 if the file doesn't exist
+     */
+    @GetMapping("files/attachments/slides/attachment-unit/{attachmentUnitId}/slide/{slideNumber}")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<byte[]> getAttachmentUnitAttachmentSlide(@PathVariable Long attachmentUnitId, @PathVariable String slideNumber) {
```

Comment:
This endpoint looks weird to me. Why don't we just use the image path that we save in the client to retrieve the image as we to in the Drag and Drop quiz? I don't think it's necessary to return the whole file in this situation, is it?
There is a reason we have the internal file path and the public one.

Reply:
This endpoint is used to get the slide which is an image. This is being requested when markdown editor is in preview mode then the image tag makes the request to this endpoint. 

This is a bit different as we dont want to have the file name in the markdown editor. Then i have to request it by slide number `api/files/attachments/slides/attachment-unit/123/slide/2` so i get the image by using attachment unit id and slide number.